spqr_engine.py

```diff
@@ -73,7 +74,23 @@ def quantize(
         ), see class QuantizationResult below for details
         """
         weight = self.layer.weight.detach().to(dtype=torch.float, copy=True)
+        save_quant_dict = {}
         perm = get_permutation_order(self.H, weight, permutation_order)
+
+        if save_quantization:
+            save_quant_dict['quant_weights'] = []
+            save_quant_dict['quant_layer_scale'] = []
+            save_quant_dict['quant_layer_zeros'] = []
+            save_quant_dict['quant_layer_scale_qq_scale'] = []
+            save_quant_dict['quant_layer_scale_qq_zero'] = []
+            save_quant_dict['quant_layer_zero_qq_scale'] = []
+            save_quant_dict['quant_layer_zero_qq_zero'] = []
+            save_quant_dict['save_float_dtype'] = torch.float16
```

Comment:
Maybe add option to choose float32? I encountered issues with numerics, when reconstructing the weights